projects/pgai/tests/vectorizer/conftest.py

```diff
@@ -113,3 +119,50 @@ def postgres_container(
     load_openai_key: bool = params.get("load_openai_key", True)  # type: ignore
 
     return postgres_container_manager(load_openai_key=load_openai_key)  # type: ignore
+
+
+class ReverseProxy:
+    def __init__(self, target_url: str):
+        self.target = urlparse(target_url)
+
+    def request(self, flow: HTTPFlow):
+        flow.request.host = str(self.target.hostname)
+        flow.request.path = self.target.path + flow.request.path
+        flow.request.scheme = "https"
+        flow.request.port = 443
+
+
+def run_reverse_proxy(
+    target_host: str, listen_port: int = 8000, listen_host: str = "localhost"
+):
+    async def start_proxy():
+        opts = options.Options(listen_host=listen_host, listen_port=listen_port)
+        master = DumpMaster(opts)
+        master.addons.add(ReverseProxy(target_host))  # type:ignore
+
+        try:
+            await master.run()
+        finally:
+            master.shutdown()
+
+    # mitmproxy relies on asyncio for its event loop
+    def run_event_loop():
+        asyncio.run(start_proxy())
+
+    proxy_thread = threading.Thread(target=run_event_loop, daemon=True)
+    proxy_thread.start()
+
+
+@pytest.fixture(scope="function")
+def openai_proxy_url(request: pytest.FixtureRequest):
```

Comment:
Is there a good reason tu use yield here ? Maybe I am missing something, but as we always return just a single value, I think using always return instead of yield would simpler.

Reply:
Not anymore. You are right.

Reply:
fixed